### DISCUSSION TITLE
Change the default location for Execute-ProcessAsUser

### DISCUSSION THREAD

**Discussion by ghost** [PSAppDeployToolkit/PSAppDeployToolkit#1071] [General]
Our IT Security team would like us to change the default location to a non-user writable location so it can be added as an applocker exclusion. Has anyone done this successfully before?

**Comment by DanGough** [comment]
Execute-ProcessAsUser has a -TempPath switch you can use to override the location.

The default location though, when running an install as system would be the temp folder of the System account - C:\Windows\Temp or C:\Windows\SystemTemp, both of which are not user writable by default.

FYI, in the forthcoming v4, no folder is used for this since the vbscript file / scheduled task approach is no longer used.

**Reply by ghost** [reply]
We've found that the default location is C:\Users\Public - I wanted to change the default for our template so that it's not possible for any of our engineers to forget add in -TempPath and have it break.

No worries if it's not as simple a simple change for us to make on our copy - I've had a bit of a look through the backend scripts and it doesn't look that way!

**Comment by DanGough** [comment]
Check AppDeployToolkitConfig.xml, the default value is:

<Toolkit_TempPath>$envTemp</Toolkit_TempPath>

**Comment by mjr4077au** [comment]
Closing this one out as it's irrelevant now that v4 is out.
